### DISCUSSION TITLE
Documentation: sign for power flow

### DISCUSSION THREAD

**Discussion by chochon** [e2nIEE/pandapower#1334] [Q&A]
Hello,
I am beginner of pandapower.
I read 'Unit system and conventions' in documentation of pandapower.

[https://pandapower.readthedocs.io/en/develop/about/units.html](url)

In the documentation, it mentioned that the power flow values for branch elements (e.g. lines & transformers) are always defined as the power flow into the branch element (positive values means power flows into the element, negative power means power flows out of the element).
At this point, I want to know deeply about 'Simple Example Network'.

[https://nbviewer.jupyter.org/github/e2nIEE/pandapower/blob/develop/tutorials/create_simple.ipynb](url)

In case of transmission line 4 of simple example network, I think power flows from bus 7 into bus 5. However, the results of power flow are different sign as following:

[https://nbviewer.jupyter.org/github/e2nIEE/pandapower/blob/develop/tutorials/powerflow.ipynb](url)

p_from_mw : 8.000000e-01
p_to_mw : -7.936182e-01

Could you please explain to understand easily for me?

**Comment by rbolgaryn** [comment]
Hi @chochon ,

try the following example:

```python
net = pp.create_empty_network()
b1, b2 = pp.create_buses(net, 2, 110)
pp.create_line(net, b1, b2, std_type= "N2XS(FL)2Y 1x300 RM/35 64/110 kV", length_km=1)
pp.create_ext_grid(net, b1)
pp.create_load(net, b2, 2)
pp.runpp(net)
net.res_line
```

You can see that the power at from_bus is positive (bus with the ext_grid) and at the to_bus is negative (at the load bus).
The positive value means the power is flowing into the line, the negative value means power is flowing out of the line.

Roman
